File: src/main/java/Sample.go
package main

import "fmt"

func main() {
	fmt.Println("This is a sample code snippet.")
}

func personFunction() {
	printWada()
	p := returnYamada()
	fmt.Println(p)
}

func printWada() {
	wada := &Person{"Wada", 26}
	fmt.Println(wada)
}

func returnYamada() *Person {
	yamada := &Person{"Yamada", 22}
	return yamada
}

type Person struct {
	name string
	age  int
}

File: src/main/java/Sample.rs
#[derive(Debug)]

fn main() {
    println!("This is a sample code snippet.");
}

fn person_function() {
    print_wada();
    let p = return_yamada();
    println!("{:?}", p);
}

fn print_wada() {
    let wada = Person {
        name: "Wada".to_string(),
        age: 26,
    };
    println!("{:?}", wada);
}

fn return_yamada() -> Person {
    let yamada = Person {
        name: "Yamada".to_string(),
        age: 22,
    };
    yamada
}

struct Person {
    name: String,
    age: i32,
}
